tensorboard/plugins/projector/vz_projector/data.ts

```diff
@@ -358,41 +358,43 @@ export class DataSet {
     const knnComputation = this.computeKnn(sampledData, k)
 
     knnComputation.then(nearest => {
-      this.nearest = nearest;
       util.runAsyncTask('Initializing T-SNE...', () => {
-            this.tsne.initDataDist(this.nearest);
-          }).then(step);
+        this.tsne.initDataDist(nearest);
+      }).then(step);
     });
   }
 
   /** Runs UMAP on the data. */
   async projectUmap(
-    nComponents: number, 
-    nNeighbors: number, 
+    nComponents: number,
+    nNeighbors: number,
     stepCallback: (iter: number) => void) {
     this.hasUmapRun = true;
-    this.umap = new UMAP({nComponents, nNeighbors});
 
-    let currentEpoch = 0;
-    const epochStepSize = 10;
     const sampledIndices = this.shuffledDataIndices.slice(0, UMAP_SAMPLE_SIZE);
-  
     const sampledData = sampledIndices.map(i => this.points[i]);
-    // TODO: Switch to a Float32-based UMAP internal
     const X = sampledData.map(x => Array.from(x.vector));
 
-    this.nearest = await this.computeKnn(sampledData, nNeighbors);
-    
+    nNeighbors = Math.min(X.length, nNeighbors)
```

Comment:
What are the implications of this change? If `X.length < nNeighbors`
(which can only happen on small datasets), did we previously crash at
some point down the pipeline, or yield wrong results, or something else?
Is this a separate bug?

Reply:
I've actually removed this from the other PR - it's essentially a noop (umap does this internally) but it seemed like a good idea at the time.